prow/config/jobs_test.go

```diff
@@ -76,8 +79,8 @@ func TestPresubmits(t *testing.T) {
 			}
 			// Check that the merge bot will run AlwaysRun jobs, otherwise it
 			// will attempt to rerun forever.
-			if job.AlwaysRun && !job.re.MatchString(testThis) {
-				t.Errorf("AlwaysRun job %s: \"%s\" does not match regex \"%v\".", job.Name, testThis, job.Trigger)
+			if job.AlwaysRun && !job.re.MatchString(newTestThis) && !job.re.MatchString(oldTestThis) {
```

Comment:
The boolean logic here is such that this test case will pass if either newTestThis or oldTestThis matches when AlwaysRun (try it out).

A cleaner way to do this might be to check:
```
`job.re.MatchString(newTestThis) == job.re.MatchString(oldTestThis)` // Do the new and old test do the same thing?
`job.AlwaysRun == job.re.MatchString(newTestThis)`  // Do we do the right thing based on AlwaysRun?
```
Although feel free to just fix up this line `if Always && !matchNew && !matchOld { error }` produces the wrong result when `matchNew=true` but `matchOld=false`